### PR TITLE
TH-173-aura: Aura > Cart drawer > Format currency based on Dotshop locale

### PR DESCRIPTION
**Pull request by eihabkhan1**
## JIRA Ticket

Ticket: [TH-173](https://youcanshop.atlassian.net/browse/TH-173).

### Multi Currency **ON**
* [ ] From the multi currency settings in seller area, turn on multi currency and enable precision
* [ ] Turn on your VPN
* [ ] Go to your product's page in storefront
* [ ] Add it to the cart
* [ ] Make sure that:
    * [ ] The price for each item in the cart is formatted based on your VPN's country + your browser's language
    * [ ] The price is converted to your VPN's country currency
    * [ ] The currency symbol displayed is the VPN's country currency symbol
    * [ ] The subtotal is also formatted accordingly
* [ ] Go to your browser's language settings, If you're using Chrome just click `⌘ + ,`
* [ ] Search for language
Screenshot 2024-10-08 at 8 43 12 AM

* [ ] Click the three dots next to any language and move it to the top to make it your language of choice
* [ ] Go back to your storefront, hit refresh and open your cart
* [ ] The price value(s) and subtotal should be formatted based on your VPN + your browser's language

### Multi Currency **OFF**
* [ ] From the multi currency settings in seller area, turn off multi currency and enable precision
* [ ] Go to your store's general settings `/admin/settings/general#details`
* [ ] Change the store language to `English - United States`
* [ ] Change the store currency to `Moroccan Dirhams`
* [ ] Change the currency symbol to `MAD`
* [ ] Go to your product page
* [ ] Add it to the cart
* [ ] Hit refresh, make sure that:
    * [ ] The price (value) is formatted based on your store country
    * [ ] The subtotal is also formatted accordingly
    * [ ] The currency symbol displayed is the symbol you defined in your store (in this case it should be `MAD`)
* [ ] Go back to your store, change the language to `French - France`
* [ ] Go back to storefront, hit refresh, and open the cart
* [ ] The price should be formatted accordingly and the currency symbol should be on the right
* [ ] Go back to your store, this time change the currency symbol to a random currency symbol (ex: `Wiwi`)
* [ ] Go back to storefront, hit refresh, and open the cart
* [ ] The price should be formatted accordingly and the currency symbol should still be on the right

## Note

Leave empty when you have nothing to say.

[TH-173]: https://youcanshop.atlassian.net/browse/TH-173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: themes/aura/assets/money.js
Comment (eihabkhan1) [inline themes/aura/assets/money.js:1]: Removed this file entirely so we don't have to include it each time we use it. Will add its content to main.js (till we start using mjs 👍 )

File: themes/aura/sections/product.liquid
Comment (eihabkhan1) [inline themes/aura/sections/product.liquid:1]: The code used in `money.js` now resides in `main.js`

File: themes/aura/assets/main.js [hunk 1]
```diff
@@ -301,3 +303,44 @@ if (FORM.errors) {
...
+
+  const determineSymbolPositionFormatter = new Intl.NumberFormat(locale, {
+    style: 'currency',
+    currency: 'USD',
+    currencyDisplay: 'symbol',
+  });
+
+  const parts = determineSymbolPositionFormatter.formatToParts(1); // format with 1 USD just to determine the position of the currency symbol
+  const symbolIndex = parts.findIndex(part => part.type === 'currency');
+  const isSymbolOnLeft = symbolIndex === 0;
```
Comment (Othman-Charai) [inline themes/aura/assets/main.js:334]: could be removed, inline check !

File: themes/aura/assets/main.js [hunk 2]
```diff
@@ -301,3 +303,44 @@ if (FORM.errors) {
...
+
+  return isSymbolOnLeft
```
Comment (Othman-Charai) [inline themes/aura/assets/main.js:336]:
```suggestion
  return symbolIndex === 0
```

**Review by Othman-Charai — APPROVED** [review]

**Pull request merged**